### PR TITLE
Stats: Add annual highlights to insights page

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Proposed Changes

image

* Adds an annual highlight component to the top of the Insights page.
* This feature is behind a feature flag, `stats/new-annual-highlights`.
Enable the feature by adding `?flags=stats/new-annual-highlights` to the URL.

#### Testing Instructions

* Spin up this change on a live branch.
* Navigate to the Insight page of the Stats section.
* Ensure you don't see the new highlight cards.
* Add `?flags=stats/new-annual-highlights` and reload the page.
* Ensure you see the new highlight cards.
* Ensure that the existing annual site stats section has been hidden.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #69227.

### CODE REVIEW

**Review by dognose24 — COMMENTED** [review]
LGTM! However, I found a few styles on `.highlight-cards` and `.highlight-card` that differ from the design.

截圖 2022-11-02 下午10 02 57

截圖 2022-11-02 下午9 32 03

If the last `HighlightCard` needs to cross over the right side edge, would it be possible to set `overflow: hidden` to the wrapper and add ellipsis styles (e.g., gradient mask) to highlight not reaching the end? That will also benefit the scrollable list on mobile devices.

File: packages/components/src/highlight-cards/style.scss [hunk 1]
```diff
@@ -30,7 +30,7 @@ $vertical-margin: 32px;
 
 	.highlight-card {
 		border-color: var(--studio-gray-5);
-		border-radius: 4px;
+		border-radius: 5px; // stylelint-disable-line scales/radii
```
Comment (dognose24) [inline packages/components/src/highlight-cards/style.scss:33]: I think leaving the `border-radius: 4px` looks good to me, which corresponds to the `stylelint` rules. 😄

File: packages/components/src/highlight-cards/style.scss [hunk 2]
```diff
@@ -24,10 +24,13 @@ $vertical-margin: 32px;
...
+	max-width: 100%;
+	overflow-x: scroll;
+	padding: 1px; // Ensures scrollable content box doesn't cover the highlight card.
 
 	.highlight-card {
 		border-color: var(--studio-gray-5);
-		border-radius: 4px;
+		border-radius: 5px; // stylelint-disable-line scales/radii
 		width: 100%;
 		min-width: 200px; // Minimum mobile width
```
Comment (dognose24) [inline packages/components/src/highlight-cards/style.scss:35]:
```suggestion
		min-width: 200px; // Minimum mobile width
		padding: 16px 24px;
		margin-bottom: 0;
```

Some styles are not precisely the same as the design, which would not be a blocker. 🚀

**Review by dognose24 — APPROVED** [review]

File: packages/components/src/highlight-cards/style.scss (hunk 1 above)
Comment (jsnmoon) [reply packages/components/src/highlight-cards/style.scss:33]: I'll leave this alone for now 🙂

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7610530

Thank you @jsnmoon for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7610530).

**Pull request merged**